content/blog/new-graphic-features/index.Rmd

```diff
@@ -0,0 +1,295 @@
+
+When it comes to graphic systems you can expect that grid will be the first, perhaps only, system that ends up supporting new features in the graphics engine. Part of the reason for that is that the grid API is more powerful in general and as new and more complex graphic powers are exposed it can be easiest to make them fit into the most expressive API. This is definitely the case for the latest batch of new features, but I also expect it to be the case going forward. Just because a functionality is exposed in grid, doesn't mean that it can easily be handled in e.g. ggplot2. I'll address what the new features may mean for the future of ggplot2 in the end of the post.
+
+For graphic devices the water is more muddled. Not all devices are under active development and those devices are unlikely to add support for new features. Further, it may be that a graphics device writes to a format, or uses a library, that do not support a new feature provided by the graphics engine. The bottom line is that we can expect an increased fragmentation of the graphics devices in R in terms of which will be up to specs with the latest graphics engine features. It appears that the cairo based devices along with the pdf device from grDevices can be expected to stay current. On our end, we will do our best to make sure that our graphic device packages (currently ragg and svglite) will stay on top of any new additions to the graphics engine.
+
+## The new features
+
+OK, so we've talked a lot about some new features without ever going into details with what they are. If you've never felt constrained by the capabilities of the graphics in R you may be forgiven for thinking that this is all a big fuzz over nothing. You may be right, but new capabilities will often allow the ecosystem to evolve in new and unexpected ways to the benefit of all.
+
+### Gradients
+
+While gradients have been a part of R graphics for a while, they have always relied on some hack - most often cutting the line or polygon in smaller bits and coloring these with color sampled from a gradient. However, now gradients are supported at the device level, meaning that the pixel color as calculated based on a gradient function. For now, the functionality is limited to fills, so if you want to draw a gradient line you still have to cut it up into small segments.
```

Comment:
Needs sentence explaining why this is an improvement.

Reply:
you mean gradients in general when they don't support strokes?

Reply:
Something like: this means instead of hundreds of small objects in the graphics file, there's now one more complicated one. This leads to considerable performance improvements, especially when outputting to vector graphics like `.pdf` and `.svg` where the file size will also be much smaller.